### PR TITLE
Support silent mode fully

### PR DESCRIPTION
**Pull request by onishiweb**
This PR adds support for full silent mode in `o-header`.

All styles are now contained within mixins to be output separately or with the newly added `oHeader` mixin.

The `oHeader` mixin will output the base styles and row styles specified (or all if none specified).

As this PR improves the support for silent mode and doesn't effect the output in non-silent mode, it shouldn't require a breaking change.

### CODE REVIEW

File: src/scss/mega/articles.scss [hunk 1]
```diff
@@ -1,18 +0,0 @@
-@if $o-header-is-silent == false {
```
Comment (onishiweb) [inline src/scss/mega/articles.scss]: @i-like-robots I've moved this and the other small partial into the main mega `menu.scss` file as there seemed no specific reason to keep them separate and the `menu.scss` file wasn't too large for it to become unwieldy. Happy to revert the change though if there was a good reason behind it?

**Comment by onishiweb** [comment]
I think the docs could be improved but I wanted to open the PR now to get people's thoughts on the implementation first. Will have another look at the docs tomorrow with fresh :eyes:

File: src/scss/mega/articles.scss (hunk 1 above)
Comment (i-like-robots) [inline src/scss/mega/articles.scss]: The original plan (I'm not aware of the status of it) was to create several more modules in addition to this.

Comment (i-like-robots) [inline src/scss/mega/articles.scss]: I think this is fine though, I can't foresee us taking on any meganav work for a while and it would only be a small change if and when we do pick it up.

Comment (onishiweb) [inline src/scss/mega/articles.scss]: Cool, I wondered if it was pre-empting something coming soon. In that case, let's leave it and optimise if/when we do the work later on. Cheers :)

**Comment by onishiweb** [comment]
@alicebartlett This is now ready to review

File: README.md [hunk 2]
```diff
@@ -68,45 +68,31 @@ will output an image logo url in the format `https://my.image.service/foo/v1/ima
 
 #### Silent mode ([docs](http://origami.ft.com/docs/syntax/scss/#silent-styles))
 
-When using `o-header` in silent mode, we offer a series of helper mixins to output styles for different parts of the header. The header is made up of various features (e.g. `Nav` and `Search`), has different configurations like `sticky` and `simple` and can be combined with additional features, such as the `drawer` and `megamenu`.
+When using `o-header` in silent mode, we offer a series of helper mixins to output styles for different parts of the header.
 
 ##### oHeader
 
-To get the default header, use the `oHeader` mixin. The `oHeader` mixin will output the base styles for the header and accepts a list of the feature styles it should also output. By default the mixin will output all the feature styles if no list is specified.
+The header is made up of various features (e.g. `Nav`, `Search`, and `Drawer`). To get everything, use the `oHeader()` mixin without arguments. To get only the stuff you need, you can pass in a list of features.
```
Comment (alicebartlett) [inline README.md]: This is much simpler and clearer :+1:

Comment (alicebartlett) [inline README.md]: here, you've capitalised `Nav` etc, but below they aren't capitalised. Should the user be passing in capitalised arguments or not?

File: README.md [hunk 3]
```diff
@@ -68,45 +68,31 @@ will output an image logo url in the format `https://my.image.service/foo/v1/ima
...
-@import 'o-header/main';
-// Outputs all base styles and feature styles
-@include oHeader;
-```
+The list of features is as follows:
 
-The available feature styles that can be output are: `nav`, `anon`, `search`, and `subnav`. To specify the features for the standard header for example you would use:
+`nav`: The primary nav, this is the nav that sits directly under the top section of the header.
+`search`: The search bar that appears when a user presses the search icon.
+`anon`: Styles for the row that appears when a user is not logged in
```
Comment (alicebartlett) [inline README.md]: (I realise that I wrote this originally but... can we have a full stop at the end of this sentence...)

File: README.md [hunk 4]
```diff
@@ -68,45 +68,31 @@ will output an image logo url in the format `https://my.image.service/foo/v1/ima
...
+`sticky`: Styles that make the header stick to the top of the page when scrolling down.
+`simple`: Styles for the reduced slimmer header for article pages and subbrand pages.
+`subbrand`: Styles for the subbranded section of the header (eg Life&Arts)
```
Comment (alicebartlett) [inline README.md]: and here .

File: src/scss/_header.scss [hunk 5]
```diff
@@ -1,29 +1,25 @@
...
 		}
-
-		&:after {
-			clear: both;
+	} @else {
+		@each $feature in $features {
+			@if index($_o-header-features, $feature) != null {
+				@include _oHeaderGetFeatureMixin($feature);
+			} @else {
+				@warn "There is no row '#{$feature}' available in o-header please use one of "
```
Comment (alicebartlett) [inline src/scss/_header.scss]: nice!

File: origami.json [hunk 6]
```diff
@@ -22,34 +22,39 @@
...
             "name": "simple-header",
+            "title": "Simple Header",
             "data": "demos/src/header.js",
             "template": "demos/src/simple-header.mustache",
             "expanded": true,
             "description": ""
         },
         {
             "name": "sticky-header",
+            "title": "Sticky Header",
```
Comment (alicebartlett) [inline origami.json]: Can these titles be sentence case (eg Sticky header). We currently have a mix in this file anyway...

**Comment by alicebartlett** [comment]
This looks good to me, I think this features approach is really neat.

File: README.md (hunk 4 above)
Comment (i-like-robots) [inline README.md]: should this be camelCase like `megaMenu` below?

File: main.scss [hunk 7]
```diff
@@ -9,21 +9,22 @@
...
+@import 'src/scss/features/anon';
+@import 'src/scss/features/top';
+@import 'src/scss/features/search';
+@import 'src/scss/features/nav';
+@import 'src/scss/features/subnav';
+@import 'src/scss/features/megamenu';
+@import 'src/scss/features/subbrand';
+@import 'src/scss/features/simple';
+@import 'src/scss/features/sticky';
+@import 'src/scss/features/drawer';
```
Comment (i-like-robots) [inline main.scss]: I like this 👍

File: origami.json [hunk 8]
```diff
@@ -22,34 +22,39 @@
...
             "data": "demos/src/header.js",
             "template": "demos/src/sticky-header.mustache",
             "expanded": true,
```
Comment (i-like-robots) [inline origami.json]: Jake told me to remove the `expanded` properties from the manifest in my last PR

File: src/scss/_header.scss (hunk 5 above)
Comment (i-like-robots) [inline src/scss/_header.scss]: Warnings are easy to ignore so maybe upgrade to `@error`?
